daemon/containerd/resolver.go

```diff
@@ -27,10 +27,7 @@ func (i *ImageService) newResolverFromAuthConfig(authConfig *registrytypes.AuthC
 func hostsWrapper(hostsFn docker.RegistryHosts, optAuthConfig *registrytypes.AuthConfig, regService RegistryConfigProvider) docker.RegistryHosts {
 	var authorizer docker.Authorizer
 	if optAuthConfig != nil {
-		auth := *optAuthConfig
-		if auth != (registrytypes.AuthConfig{}) {
-			authorizer = docker.NewDockerAuthorizer(authorizationCredsFromAuthConfig(auth))
-		}
+		authorizer = docker.NewDockerAuthorizer(authorizationCredsFromAuthConfig(*optAuthConfig))
```

Comment:
Should we add a comment somewhere to explain that it's always needed?

Reply:
It might confuse people? As is the code read well: "if the auth config is not nil, create the authorizer". 

The old code that also checked if it's empty would have needed a comment I think because it wasn't obvious why we need to check for nil _and_ empty

Reply:
Wouldn't it be more correct to ALWAYS set the authorizer, and decide whether we pass the AuthorizerOpt returned from `authorizationCredsFromAuthConfig` based on the AuthConfig?

Reply:
The current implementation will reintroduce https://github.com/moby/moby/issues/45397

Reply:
We can't if `optAuthConfig` is nil, it doesn't actually look that it's nil anyway:


https://github.com/moby/moby/blob/9e6370819bf61de7583ea00af13fcbedd2bbbfe7/api/types/registry/authconfig.go#L60

https://github.com/moby/moby/blob/9e6370819bf61de7583ea00af13fcbedd2bbbfe7/api/types/registry/authconfig.go#L84

Reply:
@vvoland It doesn't seem like it reintroduces #45397 

Pulling `hello-world` while logged-out shows me these logs:

```
DEBU[2023-05-12T10:02:22.582889639+02:00] Calling HEAD /_ping
DEBU[2023-05-12T10:02:22.583745921+02:00] Calling POST /v1.42/images/create?fromImage=hello-world&tag=latest
DEBU[2023-05-12T10:02:22.608561023+02:00] resolving                                     host=registry-1.docker.io
DEBU[2023-05-12T10:02:22.608632731+02:00] do request                                    host=registry-1.docker.io request.header.accept="application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.oci.image.manifest.v1+json, application/vnd.oci.image.index.v1+json, */*" request.header.user-agent=containerd/1.6.21+unknown request.method=HEAD url="https://registry-1.docker.io/v2/library/hello-world/manifests/latest"
DEBU[2023-05-12T10:02:23.007997174+02:00] fetch response received                       host=registry-1.docker.io response.header.content-length=162 response.header.content-type=application/json response.header.date="Fri, 12 May 2023 08:02:22 GMT" response.header.docker-distribution-api-version=registry/2.0 response.header.docker-ratelimit-source=62.35.48.5 response.header.strict-transport-security="max-age=31536000" response.header.www-authenticate="Bearer realm=\"https://auth.docker.io/token\",service=\"registry.docker.io\",scope=\"repository:library/hello-world:pull\"" response.status="401 Unauthorized" url="https://registry-1.docker.io/v2/library/hello-world/manifests/latest"
DEBU[2023-05-12T10:02:23.008268046+02:00] Unauthorized                                  header="Bearer realm=\"https://auth.docker.io/token\",service=\"registry.docker.io\",scope=\"repository:library/hello-world:pull\"" host=registry-1.docker.io
DEBU[2023-05-12T10:02:23.008418378+02:00] do request                                    host=registry-1.docker.io request.header.accept="application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.oci.image.manifest.v1+json, application/vnd.oci.image.index.v1+json, */*" request.header.user-agent=containerd/1.6.21+unknown request.method=HEAD url="https://registry-1.docker.io/v2/library/hello-world/manifests/latest"
DEBU[2023-05-12T10:02:23.462543719+02:00] fetch response received                       host=registry-1.docker.io response.header.content-length=2561 response.header.content-type=application/vnd.docker.distribution.manifest.list.v2+json response.header.date="Fri, 12 May 2023 08:02:23 GMT" response.header.docker-content-digest="sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0" response.header.docker-distribution-api-version=registry/2.0 response.header.docker-ratelimit-source=62.35.48.5 response.header.etag="\"sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0\"" response.header.ratelimit-limit="100;w=21600" response.header.ratelimit-remaining="97;w=21600" response.header.strict-transport-security="max-age=31536000" response.status="200 OK" url="https://registry-1.docker.io/v2/library/hello-world/manifests/latest"
DEBU[2023-05-12T10:02:23.462832841+02:00] resolved                                      desc.digest="sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0" host=registry-1.docker.io
DEBU[2023-05-12T10:02:23.470770462+02:00] fetch                                         digest="sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0" mediatype=application/vnd.docker.distribution.manifest.list.v2+json size=2561
DEBU[2023-05-12T10:02:23.488412934+02:00] do request                                    digest="sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0" mediatype=application/vnd.docker.distribution.manifest.list.v2+json request.header.accept="application/vnd.docker.distribution.manifest.list.v2+json, */*" request.header.user-agent=containerd/1.6.21+unknown request.method=GET size=2561 url="https://registry-1.docker.io/v2/library/hello-world/manifests/sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0"
DEBU[2023-05-12T10:02:23.815099677+02:00] fetch response received                       digest="sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0" mediatype=application/vnd.docker.distribution.manifest.list.v2+json response.header.content-length=2561 response.header.content-type=application/vnd.docker.distribution.manifest.list.v2+json response.header.date="Fri, 12 May 2023 08:02:23 GMT" response.header.docker-content-digest="sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0" response.header.docker-distribution-api-version=registry/2.0 response.header.docker-ratelimit-source=62.35.48.5 response.header.etag="\"sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0\"" response.header.ratelimit-limit="100;w=21600" response.header.ratelimit-remaining="97;w=21600" response.header.strict-transport-security="max-age=31536000" response.status="200 OK" size=2561 url="https://registry-1.docker.io/v2/library/hello-world/manifests/sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0"
DEBU[2023-05-12T10:02:23.846961076+02:00] fetch                                         digest="sha256:efebf0f7aee69450f99deafe11121afa720abed733943e50581a9dc7540689c8" mediatype=application/vnd.docker.distribution.manifest.v2+json size=525
DEBU[2023-05-12T10:02:23.851984188+02:00] do request                                    digest="sha256:efebf0f7aee69450f99deafe11121afa720abed733943e50581a9dc7540689c8" mediatype=application/vnd.docker.distribution.manifest.v2+json request.header.accept="application/vnd.docker.distribution.manifest.v2+json, */*" request.header.user-agent=containerd/1.6.21+unknown request.method=GET size=525 url="https://registry-1.docker.io/v2/library/hello-world/manifests/sha256:efebf0f7aee69450f99deafe11121afa720abed733943e50581a9dc7540689c8"
DEBU[2023-05-12T10:02:23.973383270+02:00] fetch response received                       digest="sha256:efebf0f7aee69450f99deafe11121afa720abed733943e50581a9dc7540689c8" mediatype=application/vnd.docker.distribution.manifest.v2+json response.header.content-length=525 response.header.content-type=application/vnd.docker.distribution.manifest.v2+json response.header.date="Fri, 12 May 2023 08:02:23 GMT" response.header.docker-content-digest="sha256:efebf0f7aee69450f99deafe11121afa720abed733943e50581a9dc7540689c8" response.header.docker-distribution-api-version=registry/2.0 response.header.docker-ratelimit-source=62.35.48.5 response.header.etag="\"sha256:efebf0f7aee69450f99deafe11121afa720abed733943e50581a9dc7540689c8\"" response.header.ratelimit-limit="100;w=21600" response.header.ratelimit-remaining="96;w=21600" response.header.strict-transport-security="max-age=31536000" response.status="200 OK" size=525 url="https://registry-1.docker.io/v2/library/hello-world/manifests/sha256:efebf0f7aee69450f99deafe11121afa720abed733943e50581a9dc7540689c8"
DEBU[2023-05-12T10:02:23.994642120+02:00] fetch                                         digest="sha256:b038788ddb222cb7d6025b411759e4f5abe9910486c8f98534ead97befd77dd7" mediatype=application/vnd.docker.container.image.v1+json size=1485
DEBU[2023-05-12T10:02:23.998171123+02:00] do request                                    digest="sha256:b038788ddb222cb7d6025b411759e4f5abe9910486c8f98534ead97befd77dd7" mediatype=application/vnd.docker.container.image.v1+json request.header.accept="application/vnd.docker.container.image.v1+json, */*" request.header.user-agent=containerd/1.6.21+unknown request.method=GET size=1485 url="https://registry-1.docker.io/v2/library/hello-world/blobs/sha256:b038788ddb222cb7d6025b411759e4f5abe9910486c8f98534ead97befd77dd7"
DEBU[2023-05-12T10:02:24.168963120+02:00] fetch response received                       digest="sha256:b038788ddb222cb7d6025b411759e4f5abe9910486c8f98534ead97befd77dd7" mediatype=application/vnd.docker.container.image.v1+json response.header.accept-ranges=bytes response.header.age=686409 response.header.cache-control="public, max-age=14400" response.header.cf-cache-status=HIT response.header.cf-ray=7c6123a50bf32a6f-CDG response.header.connection=keep-alive response.header.content-length=1485 response.header.content-type=application/octet-stream response.header.date="Fri, 12 May 2023 08:02:24 GMT" response.header.etag="\"4c8c8c3827591cf7643245910181ba11\"" response.header.expires="Fri, 12 May 2023 12:02:24 GMT" response.header.last-modified="Thu, 04 May 2023 08:43:33 GMT" response.header.server=cloudflare response.header.vary=Accept-Encoding response.header.x-amz-id-2="kDNqB7+tawkeCGlaL99CjxxO5LfeAgCKd6Bly8Mo9I/BcaGKRjWZvsUoQjMia0p7w9QE9b6JtF8=" response.header.x-amz-request-id=8X9SW8SYGGJV1RA4 response.header.x-amz-server-side-encryption=AES256 response.header.x-amz-version-id=UbWK6i2XuagOLBKXZZ5a3JY1SIrQleVV response.status="200 OK" size=1485 url="https://registry-1.docker.io/v2/library/hello-world/blobs/sha256:b038788ddb222cb7d6025b411759e4f5abe9910486c8f98534ead97befd77dd7"
DEBU[2023-05-12T10:02:24.201861967+02:00] fetch                                         digest="sha256:70f5ac315c5af948332962ff4678084ebcc215809506e4b8cd9e509660417205" mediatype=application/vnd.docker.image.rootfs.diff.tar.gzip size=3190
DEBU[2023-05-12T10:02:24.204589729+02:00] do request                                    digest="sha256:70f5ac315c5af948332962ff4678084ebcc215809506e4b8cd9e509660417205" mediatype=application/vnd.docker.image.rootfs.diff.tar.gzip request.header.accept="application/vnd.docker.image.rootfs.diff.tar.gzip, */*" request.header.user-agent=containerd/1.6.21+unknown request.method=GET size=3190 url="https://registry-1.docker.io/v2/library/hello-world/blobs/sha256:70f5ac315c5af948332962ff4678084ebcc215809506e4b8cd9e509660417205"
DEBU[2023-05-12T10:02:24.333436895+02:00] fetch response received                       digest="sha256:70f5ac315c5af948332962ff4678084ebcc215809506e4b8cd9e509660417205" mediatype=application/vnd.docker.image.rootfs.diff.tar.gzip response.header.accept-ranges=bytes response.header.age=686409 response.header.cache-control="public, max-age=14400" response.header.cf-cache-status=HIT response.header.cf-ray=7c6123a60cc72a6f-CDG response.header.connection=keep-alive response.header.content-length=3190 response.header.content-type=application/octet-stream response.header.date="Fri, 12 May 2023 08:02:24 GMT" response.header.etag="\"77c5682f99e7f7e214256ec3fec2d80e\"" response.header.expires="Fri, 12 May 2023 12:02:24 GMT" response.header.last-modified="Thu, 04 May 2023 08:43:30 GMT" response.header.server=cloudflare response.header.vary=Accept-Encoding response.header.x-amz-id-2="7oSha/0hqZvM8qYxKdSAcoW/ZJzZf82ZytX8QhF9QtUHS4kkRPUC5qXay5mY23CDJd6dppsHz7Q=" response.header.x-amz-request-id=DT5M7981WMA3MBX1 response.header.x-amz-server-side-encryption=AES256 response.header.x-amz-version-id=90uK7t_hJMfKcra6WhJnrRqgn1IaMC3b response.status="200 OK" size=3190 url="https://registry-1.docker.io/v2/library/hello-world/blobs/sha256:70f5ac315c5af948332962ff4678084ebcc215809506e4b8cd9e509660417205"
DEBU[2023-05-12T10:02:24.365164005+02:00] image unpacked                                chainID="sha256:a7866053acacfefb68912a8916b67d6847c12b51949c6b8a5580c6609c08ae45" config="sha256:b038788ddb222cb7d6025b411759e4f5abe9910486c8f98534ead97befd77dd7"
INFO[2023-05-12T10:02:24.367657603+02:00] image pulled                                  digest="sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0" remote="docker.io/library/hello-world:latest"
```

Reply:
Could you try if it works with cURL? Wondering if there's bits that the cli handles that's not done in other cases

Reply:
(I forgot if the other issue was reported with the CLI or another client)

Reply:
With curl:

```
DEBU[2023-05-12T10:16:13.229744346+02:00] resolving                                     host=registry-1.docker.io
DEBU[2023-05-12T10:16:13.229794680+02:00] do request                                    host=registry-1.docker.io request.header.accept="application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.oci.image.manifest.v1+json, application/vnd.oci.image.index.v1+json, */*" request.header.user-agent=containerd/1.6.21+unknown request.method=HEAD url="https://registry-1.docker.io/v2/library/hello-world/manifests/latest"
DEBU[2023-05-12T10:16:13.674409118+02:00] fetch response received                       host=registry-1.docker.io response.header.content-length=162 response.header.content-type=application/json response.header.date="Fri, 12 May 2023 08:16:13 GMT" response.header.docker-distribution-api-version=registry/2.0 response.header.docker-ratelimit-source=62.35.48.5 response.header.strict-transport-security="max-age=31536000" response.header.www-authenticate="Bearer realm=\"https://auth.docker.io/token\",service=\"registry.docker.io\",scope=\"repository:library/hello-world:pull\"" response.status="401 Unauthorized" url="https://registry-1.docker.io/v2/library/hello-world/manifests/latest"
DEBU[2023-05-12T10:16:13.674764702+02:00] Unauthorized                                  header="Bearer realm=\"https://auth.docker.io/token\",service=\"registry.docker.io\",scope=\"repository:library/hello-world:pull\"" host=registry-1.docker.io
DEBU[2023-05-12T10:16:13.674910619+02:00] do request                                    host=registry-1.docker.io request.header.accept="application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.oci.image.manifest.v1+json, application/vnd.oci.image.index.v1+json, */*" request.header.user-agent=containerd/1.6.21+unknown request.method=HEAD url="https://registry-1.docker.io/v2/library/hello-world/manifests/latest"
DEBU[2023-05-12T10:16:14.541149961+02:00] fetch response received                       host=registry-1.docker.io response.header.content-length=2561 response.header.content-type=application/vnd.docker.distribution.manifest.list.v2+json response.header.date="Fri, 12 May 2023 08:16:14 GMT" response.header.docker-content-digest="sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0" response.header.docker-distribution-api-version=registry/2.0 response.header.docker-ratelimit-source=62.35.48.5 response.header.etag="\"sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0\"" response.header.ratelimit-limit="100;w=21600" response.header.ratelimit-remaining="96;w=21600" response.header.strict-transport-security="max-age=31536000" response.status="200 OK" url="https://registry-1.docker.io/v2/library/hello-world/manifests/latest"
DEBU[2023-05-12T10:16:14.541483211+02:00] resolved                                      desc.digest="sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0" host=registry-1.docker.io
DEBU[2023-05-12T10:16:14.553085854+02:00] fetch                                         digest="sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0" mediatype=application/vnd.docker.distribution.manifest.list.v2+json size=2561
DEBU[2023-05-12T10:16:14.569229546+02:00] do request                                    digest="sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0" mediatype=application/vnd.docker.distribution.manifest.list.v2+json request.header.accept="application/vnd.docker.distribution.manifest.list.v2+json, */*" request.header.user-agent=containerd/1.6.21+unknown request.method=GET size=2561 url="https://registry-1.docker.io/v2/library/hello-world/manifests/sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0"
DEBU[2023-05-12T10:16:14.879545235+02:00] fetch response received                       digest="sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0" mediatype=application/vnd.docker.distribution.manifest.list.v2+json response.header.content-length=2561 response.header.content-type=application/vnd.docker.distribution.manifest.list.v2+json response.header.date="Fri, 12 May 2023 08:16:14 GMT" response.header.docker-content-digest="sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0" response.header.docker-distribution-api-version=registry/2.0 response.header.docker-ratelimit-source=62.35.48.5 response.header.etag="\"sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0\"" response.header.ratelimit-limit="100;w=21600" response.header.ratelimit-remaining="96;w=21600" response.header.strict-transport-security="max-age=31536000" response.status="200 OK" size=2561 url="https://registry-1.docker.io/v2/library/hello-world/manifests/sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0"
DEBU[2023-05-12T10:16:14.907173278+02:00] fetch                                         digest="sha256:efebf0f7aee69450f99deafe11121afa720abed733943e50581a9dc7540689c8" mediatype=application/vnd.docker.distribution.manifest.v2+json size=525
DEBU[2023-05-12T10:16:14.912665953+02:00] do request                                    digest="sha256:efebf0f7aee69450f99deafe11121afa720abed733943e50581a9dc7540689c8" mediatype=application/vnd.docker.distribution.manifest.v2+json request.header.accept="application/vnd.docker.distribution.manifest.v2+json, */*" request.header.user-agent=containerd/1.6.21+unknown request.method=GET size=525 url="https://registry-1.docker.io/v2/library/hello-world/manifests/sha256:efebf0f7aee69450f99deafe11121afa720abed733943e50581a9dc7540689c8"
DEBU[2023-05-12T10:16:15.166872138+02:00] fetch response received                       digest="sha256:efebf0f7aee69450f99deafe11121afa720abed733943e50581a9dc7540689c8" mediatype=application/vnd.docker.distribution.manifest.v2+json response.header.content-length=525 response.header.content-type=application/vnd.docker.distribution.manifest.v2+json response.header.date="Fri, 12 May 2023 08:16:15 GMT" response.header.docker-content-digest="sha256:efebf0f7aee69450f99deafe11121afa720abed733943e50581a9dc7540689c8" response.header.docker-distribution-api-version=registry/2.0 response.header.docker-ratelimit-source=62.35.48.5 response.header.etag="\"sha256:efebf0f7aee69450f99deafe11121afa720abed733943e50581a9dc7540689c8\"" response.header.ratelimit-limit="100;w=21600" response.header.ratelimit-remaining="95;w=21600" response.header.strict-transport-security="max-age=31536000" response.status="200 OK" size=525 url="https://registry-1.docker.io/v2/library/hello-world/manifests/sha256:efebf0f7aee69450f99deafe11121afa720abed733943e50581a9dc7540689c8"
DEBU[2023-05-12T10:16:15.190582425+02:00] fetch                                         digest="sha256:b038788ddb222cb7d6025b411759e4f5abe9910486c8f98534ead97befd77dd7" mediatype=application/vnd.docker.container.image.v1+json size=1485
DEBU[2023-05-12T10:16:15.195648974+02:00] do request                                    digest="sha256:b038788ddb222cb7d6025b411759e4f5abe9910486c8f98534ead97befd77dd7" mediatype=application/vnd.docker.container.image.v1+json request.header.accept="application/vnd.docker.container.image.v1+json, */*" request.header.user-agent=containerd/1.6.21+unknown request.method=GET size=1485 url="https://registry-1.docker.io/v2/library/hello-world/blobs/sha256:b038788ddb222cb7d6025b411759e4f5abe9910486c8f98534ead97befd77dd7"
DEBU[2023-05-12T10:16:15.593071215+02:00] fetch response received                       digest="sha256:b038788ddb222cb7d6025b411759e4f5abe9910486c8f98534ead97befd77dd7" mediatype=application/vnd.docker.container.image.v1+json response.header.accept-ranges=bytes response.header.age=687240 response.header.cache-control="public, max-age=14400" response.header.cf-cache-status=HIT response.header.cf-ray=7c6137f15d740232-CDG response.header.connection=keep-alive response.header.content-length=1485 response.header.content-type=application/octet-stream response.header.date="Fri, 12 May 2023 08:16:15 GMT" response.header.etag="\"4c8c8c3827591cf7643245910181ba11\"" response.header.expires="Fri, 12 May 2023 12:16:15 GMT" response.header.last-modified="Thu, 04 May 2023 08:43:33 GMT" response.header.server=cloudflare response.header.vary=Accept-Encoding response.header.x-amz-id-2="kDNqB7+tawkeCGlaL99CjxxO5LfeAgCKd6Bly8Mo9I/BcaGKRjWZvsUoQjMia0p7w9QE9b6JtF8=" response.header.x-amz-request-id=8X9SW8SYGGJV1RA4 response.header.x-amz-server-side-encryption=AES256 response.header.x-amz-version-id=UbWK6i2XuagOLBKXZZ5a3JY1SIrQleVV response.status="200 OK" size=1485 url="https://registry-1.docker.io/v2/library/hello-world/blobs/sha256:b038788ddb222cb7d6025b411759e4f5abe9910486c8f98534ead97befd77dd7"
DEBU[2023-05-12T10:16:15.623646762+02:00] fetch                                         digest="sha256:70f5ac315c5af948332962ff4678084ebcc215809506e4b8cd9e509660417205" mediatype=application/vnd.docker.image.rootfs.diff.tar.gzip size=3190
DEBU[2023-05-12T10:16:15.627115851+02:00] do request                                    digest="sha256:70f5ac315c5af948332962ff4678084ebcc215809506e4b8cd9e509660417205" mediatype=application/vnd.docker.image.rootfs.diff.tar.gzip request.header.accept="application/vnd.docker.image.rootfs.diff.tar.gzip, */*" request.header.user-agent=containerd/1.6.21+unknown request.method=GET size=3190 url="https://registry-1.docker.io/v2/library/hello-world/blobs/sha256:70f5ac315c5af948332962ff4678084ebcc215809506e4b8cd9e509660417205"
DEBU[2023-05-12T10:16:15.743734573+02:00] fetch response received                       digest="sha256:70f5ac315c5af948332962ff4678084ebcc215809506e4b8cd9e509660417205" mediatype=application/vnd.docker.image.rootfs.diff.tar.gzip response.header.accept-ranges=bytes response.header.age=687240 response.header.cache-control="public, max-age=14400" response.header.cf-cache-status=HIT response.header.cf-ray=7c6137f26e800232-CDG response.header.connection=keep-alive response.header.content-length=3190 response.header.content-type=application/octet-stream response.header.date="Fri, 12 May 2023 08:16:15 GMT" response.header.etag="\"77c5682f99e7f7e214256ec3fec2d80e\"" response.header.expires="Fri, 12 May 2023 12:16:15 GMT" response.header.last-modified="Thu, 04 May 2023 08:43:30 GMT" response.header.server=cloudflare response.header.vary=Accept-Encoding response.header.x-amz-id-2="7oSha/0hqZvM8qYxKdSAcoW/ZJzZf82ZytX8QhF9QtUHS4kkRPUC5qXay5mY23CDJd6dppsHz7Q=" response.header.x-amz-request-id=DT5M7981WMA3MBX1 response.header.x-amz-server-side-encryption=AES256 response.header.x-amz-version-id=90uK7t_hJMfKcra6WhJnrRqgn1IaMC3b response.status="200 OK" size=3190 url="https://registry-1.docker.io/v2/library/hello-world/blobs/sha256:70f5ac315c5af948332962ff4678084ebcc215809506e4b8cd9e509660417205"
DEBU[2023-05-12T10:16:15.778144252+02:00] image unpacked                                chainID="sha256:a7866053acacfefb68912a8916b67d6847c12b51949c6b8a5580c6609c08ae45" config="sha256:b038788ddb222cb7d6025b411759e4f5abe9910486c8f98534ead97befd77dd7"
INFO[2023-05-12T10:16:15.780714172+02:00] image pulled                                  digest="sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0" remote="docker.io/library/hello-world:latest"
```